src/main/java/com/marcusslover/plus/lib/sound/Music.java

```diff
@@ -0,0 +1,213 @@
+package com.marcusslover.plus.lib.sound;
+
+import com.marcusslover.plus.lib.common.ISendable;
+import com.marcusslover.plus.lib.server.ServerUtils;
+import com.marcusslover.plus.lib.task.Task;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
+import net.kyori.adventure.audience.Audience;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Music can be used to loop notes for an audience or command sender.
+ *
+ * <p>Example:
+ * <pre>
+ *     {@code
+ *     Music.of(Note.of(Key.key("music.loop"), 0.2f, 1f), 2400).send(audience);
+ *     }
+ * </pre>
+ * This will play the note "music.loop" for 2400 ticks (2 minutes) for the audience.
+ * <p>
+ */
+@Data
+@Accessors(fluent = true)
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class Music implements ISendable<Music> {
+	protected final Note intro;
+	protected final long introLength;
+	protected final Note loop;
+	protected final long loopLength;
+	protected final Note tail;
+
+	protected Task loopTask = null;
+	public long ticks = 0;
+	protected Task tickTask = Task.asyncRepeating(ServerUtils.getCallingPlugin(), () -> {
+		ticks++;
+	}, 0, 1);
```

Comment:
Initial values should be set in constructor. Lenient to generic and nulls variables.